datafusion/core/src/datasource/listing_table_factory.rs

```diff
@@ -229,8 +229,14 @@ impl TableProviderFactory for ListingTableFactory {
         let config = ListingTableConfig::new(table_path)
             .with_listing_options(options)
             .with_schema(resolved_schema);
-        let table =
-            ListingTable::try_new(config)?.with_definition(cmd.definition.clone());
+        let provider;
```

Comment:
If you implement the `with_cache` API, this can look like

```rust
let provider = ListingTable::try_new(config)?
  .with_cache(state.runtime_env().cache_manager.get_file_statistic_cache())
```

Reply:
Move clear way ! 👍